dynawo/sources/Modeler/ModelManager/DYNModelManager.cpp

```diff
@@ -1435,21 +1434,17 @@ ModelManager::computeDelay(int exprNumber, double exprValue, double time, double
     return exprValue;
   }
 
-  if (time < delayTime || doubleEquals(time, delayTime)) {
-    // requested time is allowed but the simulation has not run for enought time to compute the delayed value:
-    // use the initial value to avoid artefacts
-    const boost::optional<double>& initialValue = delayManager_.getInitialValue(exprNumber);
-#if _DEBUG_
-    // shouldn't happen by construction
-    assert(initialValue.is_initialized());
-#endif
+  const Delay& delay = delayManager_.getDelayById(exprNumber);
+  delayManager_.getNonCstDelayById(exprNumber).setDelayTime(delayTime);
```

Comment:
I really don't get why this is necessary

Reply:
To handle a change in the delay time parameter

Reply:
I guess we will need to discuss so that I can understand. Anyway the computeDelay is probably not a good place to modify the delay

Reply:
This is the only moment where delayTime is updated from the Modelica model when it changes like in of the nrt